src/Simplify.cpp

```diff
@@ -1444,23 +1470,48 @@ class Simplify : public IRMutator {
             mod_rem = modulus_remainder(ramp_a->base, alignment_info);
         }
 
-        if (is_zero(a) && !is_zero(b)) {
+        if (is_indeterminate_expression(call_a)) { 
             expr = a;
+        } else if (is_indeterminate_expression(call_b)) {
+            expr = b;
+        } else if (is_zero(b)) {
```

Comment:
I don't think we should ever generate indeterminate expressions for floats. NaN, Inf, etc already capture that meaning, and often people want them plumbed through. We should only be catching integer divide-by-zero. For floats, we're -fpmath-equivalent, so I don't mind if we simplify x/x to 1.0f, even though x may turn out to be NaN or Inf at runtime.